upb/encode.c

```diff
@@ -258,7 +258,7 @@ static bool upb_encode_scalarfield(upb_encstate *e, const char *field_mem,
                                    bool is_proto3) {
 #define CASE(ctype, type, wire_type, encodeval) do { \
   ctype val = *(ctype*)field_mem; \
-  if (is_proto3 && val == 0) { \
+  if (is_proto3 && f->oneof_index == UPB_NOT_IN_ONEOF && val == 0) { \
```

Comment:
could factor out
const bool omit_default_values = is_proto3 && f->oneof_index == UPB_NOT_IN_ONEOF;
and change all the ifs to use it.
Maybe there is a better variable name but I think it makes it easier to read